src/persist-client/src/operators/shard_source.rs

```diff
@@ -179,9 +201,10 @@ where
     V: Debug + Codec,
     D: Semigroup + Codec64 + Send + Sync,
     F: FnMut(&PartStats) -> bool + 'static,
-    G: Scope,
+    G: Scope<Timestamp = R>,
```

Comment:
Why do we need to handle the refined timestamp in this function? It sounds like this can stay as-is and then we can call `stream.enter()` to bring it into the refined scope. Am I missing something?

EDIT: I saw I said the same thing in this comment https://github.com/MaterializeInc/materialize/pull/19758/files#r1222236224. I think this can be simplified

Reply:
`stream.enter()` is only implemented on `Child` specifically, not `: Scope`, so this changes a bunch of other nonsense